src/lib.rs

```diff
@@ -480,11 +480,11 @@ pub enum Error {
 // Passthrough Debug to Display, since errors should be user-visible
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
-        f.write_str(&self.to_string())
+        f.write_str(self.description())
     }
 }
 
-impl error::Error for Error {
+impl Error {
```

Comment:
Are you sure about that?   cause & description  suppose belong to trait. It is not just methods. If you do impl Error, that will be regular methods of this struct and it is not what expected.

Reply:
Ah, you're right!  I need keep the behavior of Error trait.